### ISSUE TITLE
Commit e10f3f8 breaks sylius-webshop, but is maybe a generall bug 

### ISSUE THREAD

**Issue by ARoddis** [symfony-cmf/Routing#137]
Hi,

the mentioned commit introduces an event for route-generation:

```php
public function generate($name, $parameters = array(), $referenceType = false)
    {
        if ($this->eventDispatcher) {
            $event = new RouterGenerateEvent($name, $parameters, $referenceType);
            $this->eventDispatcher->dispatch(Events::PRE_DYNAMIC_GENERATE, $event);
            $name = $event->getName();
            $parameters = $event->getParameters();
            $referenceType = $event->getReferenceType();
        }

        return $this->getGenerator()->generate($name, $parameters, $referenceType);
    }
```

The problem is the eventName. The EventInstance gets the parameter $name as its name in the contructor. But $name can be an object in a generator implementing the "VersatileGeneratorInterface".

The code above then overrides effectively the object with a string. The string representing the eventname. That implicitly happens in the EventDispatcher...which by itself overrides the Name of the EventInstances with the passed EventName:

```php
 /**
     * @see EventDispatcherInterface::dispatch()
     *
     * @api
     */
    public function dispatch($eventName, Event $event = null)
    {
        if (null === $event) {
            $event = new Event();
        }

        $event->setDispatcher($this);
        $event->setName($eventName);

        if (!isset($this->listeners[$eventName])) {
            return $event;
        }

        $this->doDispatch($this->getListeners($eventName), $eventName, $event);

        return $event;
    }
```

So the object in the parameter $name will get lost and is not reachable anymore...that breaks the sylius-webshop for example. 

It seems like a bug to me...because everywere else the interface "VersatileGeneratorInterface" is checked before handling the name.

Hope that was understandable. ^^ 

Im not that good in proper technical english. :)

**Comment by dbu** [comment]
yeah, that is a bug. but i think this was fixed recently in #135 - can you check if that fixes your issue?

**Comment by ARoddis** [comment]
Perfect, yes. :) Thank you @dbu
